src/primitives.coffee

```diff
@@ -85,6 +85,19 @@ class Collection extends ElementType
 
   add: (val) -> @push val
 
+  findElements: (cond) ->
+    elements = @content.reduce (results, el) ->
+      return results.concat([el.findElements(cond)]) if el.elementType() in ['array', 'object']
+      return results.concat([el]) if cond(el)
+      return results
+    , []
+    _.flatten elements, true
+
+  find: (cond) ->
```

Comment:
I see. I think my tests were not descriptive, so I'll change that. This is creating an array of elements that pass the conditional function. It pulls out the entire element, not just the values. This is useful for searching for IDs or attributes in an entire document.